File: ccbusDemo/server/ccbusdemo/src/main/java/ccbus/demo/ccbus/web/components/EcmaComponentWorker.java
package ccbus.demo.ccbus.web.components;


import ccbus.connect.core.ccbus.Error;
import ccbus.connect.core.ccbus.ExtComponent;
import ccbus.connect.core.ccbus.WorkerServer;
import ccbus.connect.core.ccbus.components.Pager;
import ccbus.connect.ecma.Date;
import ccbus.connect.system.Prop;
import ccbus.connect.system.PropEnd;
import ccbus.connect.system.State;
import ccbus.connect.system.StateEnd;

import ccbus.demo.ccbus.payload.SomePayload;
import ccbus.connect.core.ccbus.payload.PageData;
import ccbus.demo.ccbus.payload.packp.OtherPayload;
import org.joda.time.DateTime;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.function.BiFunction;
import java.util.function.Function;
import ccbus.demo.ccbus.web.components.EcmaInner;
import ccbus.demo.ccbus.service.ServiceSome;


public class EcmaComponentWorker extends Pager
{
    @State
    String name="Some Vla";
    ArrayList<String> a=new ArrayList<>();
    ArrayList<Integer> aa=new ArrayList<>();
    @StateEnd

    Date date=new Date("2012-01-02 13:00");
    DateTime dateTime=new DateTime("2012-01-02 13:00");

    //ArrayList<String> av=new ArrayList<>();
    String nameB="Some Vla";
    SomePayload payload=new SomePayload();
    OtherPayload payloadB=new OtherPayload();
    public BiFunction<Object,Object,String> handleOnChangeBi;
    public Function<Integer,Integer> handleOnChangeD=(Integer b) -> {return b+1;};

    int someInteger;

    public EcmaComponentWorker()
    {
        super();
    }


    @Override
    public void componentDidMount()
    {
        run();
    }

    public String some(Object a, Object b)
    {
        return "";
    }
    public String someS(Object a)
    {
        return "";
    }

    public void run()
   {

       this.page=2;
       this.stateKey("name","Some new val");
      this.handleOnChangeD.apply(1);
       handleOnChangeBi=this::some;
       handleOnChangeBi.apply(new Object(),payload);

       ServiceSome serviceSome=new ServiceSome();
       serviceSome.compute(this::someS,this::someS);

        HashMap<String,String> b=new HashMap<>();
      new WorkerServer<String>()
        {
            Object bind=this;
            String nameB=name;
            // Server thread
            public void compute()
            {
                a.add("My Name");
                a.add("My Name2");
                a.add(payload.toString());
                a.add(dateTime.toString());
                a.add(String.valueOf(page));
                name="New name Polya";
                complete(name);
            }

            // UI thread
            public void onCompletion(String value)
            {
                stateKey("name",value);
            }

            public void onError(Error error)
            {

            }

        };

//       new WorkerStorage<String>()
//       {
//
//           // Server thread
//           public void key()
//           {
//               this.key(payload.name);
//           }
//           // UI thread
//           public void onCompletion(String value)
//           {
//               stateKey("name",value);
//           }
//
//           public void onError(Error error)
//           {
//
//           }
//
//       };
//
//       new WorkerCache<String>()
//       {
//
//           // Server thread
//           public void key()
//           {
//               this.key("20/02/2012"+name);
//           }
//
//           // Server thread
//           public void compute()
//           {
//               a.add("My Name");
//               a.add("My Name2");
//               a.add(payload.toString());
//               a.add(dateTime.toString());
//               a.add(String.valueOf(page));
//               name="New name Polya";
//               complete(name);
//           }
//
//           // UI thread
//           public void onCompletion(String value)
//           {
//               stateKey("name",value);
//           }
//
//           public void onError(Error error)
//           {
//
//           }
//
//       };

        new WorkerServer<ArrayList>()
        {
            // Server thread
            public void compute()
            {
                String some=name;
                ArrayList<Integer> result= new ArrayList<>();
                result.addAll(Arrays.asList(1,2,3,4,5));
                complete(result);
            }

            // UI thread
            public void onCompletion(ArrayList value)
            {
                stateKey("aa",value);
            }

            public void onError(Error error)
            {

            }
        };
    }
}

File: ccbusDemo/client/ccbusdemo/src/components/EcmaComponentWorker.js
import {Error} from 'ccbus';
import {WorkerServer} from 'ccbus';
import {Pager} from 'ccbus/components/Pager';
import {SomePayload} from 'payload/SomePayload';
import {OtherPayload} from 'payload/packp/OtherPayload';
import {EcmaInner} from 'components/EcmaInner';
import React from 'react';

import {ExtComponent} from 'ccbus';
import {PageData} from 'ccbus/payload/PageData';

import {ServiceSome} from 'service/ServiceSome';

export class EcmaComponentWorker extends Pager
{
    
    /**
    * @returns {}
    */ 
    constructor()
    {
        super();
    }
    
    /**
    * @type {Object}
    */ 
    state = {
        
        /**
        * @type {String}
        */ 
        name: "Some Vla",
        
        /**
        * @type {Array<String>}
        */ 
        a: new Array(),
        
        /**
        * @type {Array<Integer>}
        */ 
        aa: new Array()
    };
    
    /**
    * @type {Date}
    */ 
    date = new Date("2012-01-02 13:00");
    
    /**
    * @type {Date}
    */ 
    dateTime = new Date("2012-01-02 13:00");
    
    /**
    * @type {String}
    */ 
    nameB = "Some Vla";
    
    /**
    * @type {SomePayload}
    */ 
    payload = new SomePayload();
    
    /**
    * @type {OtherPayload}
    */ 
    payloadB = new OtherPayload();
    
    /**
    * @type {BiFunction<Object,Object,String>}
    */ 
    handleOnChangeBi;
    
    /**
    * @type {Function<Integer,Integer>}
    */ 
    handleOnChangeD = (b) => 
    {
        return b+1;
    };
    
    /**
    * @type {number}
    */ 
    someInteger;
    
    /**
    * @returns {void}
    */ 
    componentDidMount()
    {
        this.run();
    }
    
    /**
    * @param {Object} a
    * @param {Object} b
    * @returns {String}
    */ 
    some(a,b)
    {
        return "";
    }
    
    /**
    * @param {Object} a
    * @returns {String}
    */ 
    someS(a)
    {
        console.log("SC");
        console.log(a);
        return "";
    }
    
    /**
    * @returns {void}
    */ 
    run()
    {
        this.page=2;
        this.stateKey("name","Some new val");
        this.handleOnChangeD(1);
        this.handleOnChangeBi=this.some;
        this.handleOnChangeBi(new Object(),this.payload);
        
        /**
        * @type {ServiceSome}
        */ 
        let serviceSome = new ServiceSome();
        serviceSome.compute(this.someS,this.someS);
        
        /**
        * @type {Map<String,String>}
        */ 
        let b = new Map();
        WorkerServer.compute({
            onError: function(error)
            {
            },
            onCompletion: function(value)
            {
                console.log("C");
                console.log(value);
                this.stateKey("name",value);
            },
            
            /**
            * @type {String}
            */ 
            endpoint: "EcmaComponentWorker_run_0",
            
            /**
            * @type {Object}
            */ 
            data: {
                
                /**
                * @type {Array<>}
                */ 
                a: this.state.a,
                
                /**
                * @type {SomePayload}
                */ 
                payload: this.payload,
                
                /**
                * @type {Date}
                */ 
                dateTime: this.dateTime,
                
                /**
                * @type {number}
                */ 
                page: this.page,
                
                /**
                * @type {String}
                */ 
                name: this.state.name
            },
            
            /**
            * @type {Object}
            */ 
            bind: this
        });
        WorkerServer.compute({
            onError: function(error)
            {
            },
            onCompletion: function(value)
            {
                this.stateKey("aa",value);
            },
            
            /**
            * @type {String}
            */ 
            endpoint: "EcmaComponentWorker_run_1",
            
            /**
            * @type {Object}
            */ 
            data: {
                
                /**
                * @type {String}
                */ 
                name: this.state.name
            },
            
            /**
            * @type {Object}
            */ 
            bind: this
        });
    }
    
    render()
    {
        return(
            <div>
                <div>Hello component. Requestd name: {this.state.name}</div>
                <EcmaInner propName={this.state.name}/>
                <div>
                    {this.state.aa.map(value => (
                        <span>{value},</span>
                    ))}
                </div>
            </div>
        );
    }
}